pyfr/integrators/dual/phys/controllers.py

```diff
@@ -47,8 +55,23 @@ def advance_to(self, t):
             raise ValueError('Advance time is in the past')
 
         while self.tcurr < t:
+            if self.tcurr + self._dt <= t < self.tcurr + self.i*self._dt:
+                if not self._if_near:
+                    self._dt_near = (t-self.tcurr)/((t-self.tcurr)//self._dt + 1.0)
+                    self._if_near = True
+                dt = self._dt_near
+            elif self.tcurr < t <= self.tcurr + self._dt:
+                dt = t - self.tcurr
+            else:
+                dt = self._dt
+                self._if_near = False
+
+            if self.pseudointegrator.dt != dt:
+                # Change dt in pseudo-integrator (and multi-p levels)
+                self.pseudointegrator.dt = dt
+
             # Take the physical step
-            self.step(self.tcurr, self._dt)
```

Comment:
I still think it is worth passing dt here as it gives a nicer interface.

Reply:
Could you explain more on that? I am not sure what you mean by a nicer interface. I have used self._dt in many places, so I will change accordingly.

Reply:
But it is confusing.  We have an argument `dt` to `_accept_step` but instead are using `self._dt`. What if these values are different?

Reply:
I had first tried to use a local variable `dt`. Since `self._dt` also occurs in multip.py as `self.pintgs[l]._dt = self._dt`, I deicided to keep the `self._dt` for consistency.

Reply:
The case of differing `dt` and `self._dt` would occur when we are adding adaptive physical time stepping, in which case we will have to rewind to a few time-steps before `tcurr`. Errors slowly build up, so there wouldn't be much point to rejecting one step the way we do for explicit time stepping.
So maybe we could decide how to handle the variables then?